### PR TITLE
Add an faq about async commit or 1PC not working

### PR DESCRIPTION
**Pull request by sticnarf**
Signed-off-by: Yilin Chen <sticnarf@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add an FAQ about why async commit or 1PC does not take effect.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@sticnarf 请拉一个技术 review。

**Comment by sticnarf** [comment]
PTAL @youjiali1995

**Review by youjiali1995 — APPROVED** [review]
LGTM

File: faq/deploy-and-maintain-faq.md [hunk 1]
```diff
@@ -229,6 +229,13 @@ TiDB 目前社区非常活跃，在 1.0 GA 版本发布后，还在不断的优
 
 这是因为两者计算的角度不一样。`information_schema.tables.data_length` 是通过统计信息（平均每行的大小）得到的估算值。TiKV 监控面板上的 store size 是单个 TiKV 实例的数据文件（RocksDB 的 SST 文件）的大小总和。由于多版本和 TiKV 会压缩数据，所以两者显示的大小不一样。
 
+#### 为什么事务没有使用异步提交或一阶段提交？
+
+下面几种情况下，即使通过系统变量开启了[异步提交](/system-variables.md#tidb_enable_async_commit-从-v50-版本开始引入)和[一阶段提交](/system-variables.md#tidb_enable_1pc-从-v50-版本开始引入)，也不会使用对应特性：
```
Comment (TomShawn) [inline faq/deploy-and-maintain-faq.md:234]:
```suggestion
在以下情况中，即使通过系统变量开启了[异步提交](/system-variables.md#tidb_enable_async_commit-从-v50-版本开始引入)和[一阶段提交](/system-variables.md#tidb_enable_1pc-从-v50-版本开始引入)，TiDB 也不会使用这些特性：
```

File: faq/deploy-and-maintain-faq.md [hunk 2]
```diff
@@ -229,6 +229,13 @@ TiDB 目前社区非常活跃，在 1.0 GA 版本发布后，还在不断的优
 
 这是因为两者计算的角度不一样。`information_schema.tables.data_length` 是通过统计信息（平均每行的大小）得到的估算值。TiKV 监控面板上的 store size 是单个 TiKV 实例的数据文件（RocksDB 的 SST 文件）的大小总和。由于多版本和 TiKV 会压缩数据，所以两者显示的大小不一样。
 
+#### 为什么事务没有使用异步提交或一阶段提交？
+
+下面几种情况下，即使通过系统变量开启了[异步提交](/system-variables.md#tidb_enable_async_commit-从-v50-版本开始引入)和[一阶段提交](/system-variables.md#tidb_enable_1pc-从-v50-版本开始引入)，也不会使用对应特性：
+
+- 由于 TiDB Binlog 的原理限制，使用 TiDB Binlog 时，不会使用异步提交或一阶段提交特性。
```
Comment (TomShawn) [inline faq/deploy-and-maintain-faq.md:236]:
```suggestion
- 使用 TiDB Binlog 时，受 TiDB Binlog 的实现原理限制，TiDB 不会使用异步提交或一阶段提交特性。
```

File: faq/deploy-and-maintain-faq.md [hunk 3]
```diff
@@ -229,6 +229,13 @@ TiDB 目前社区非常活跃，在 1.0 GA 版本发布后，还在不断的优
 
 这是因为两者计算的角度不一样。`information_schema.tables.data_length` 是通过统计信息（平均每行的大小）得到的估算值。TiKV 监控面板上的 store size 是单个 TiKV 实例的数据文件（RocksDB 的 SST 文件）的大小总和。由于多版本和 TiKV 会压缩数据，所以两者显示的大小不一样。
 
+#### 为什么事务没有使用异步提交或一阶段提交？
+
+下面几种情况下，即使通过系统变量开启了[异步提交](/system-variables.md#tidb_enable_async_commit-从-v50-版本开始引入)和[一阶段提交](/system-variables.md#tidb_enable_1pc-从-v50-版本开始引入)，也不会使用对应特性：
+
+- 由于 TiDB Binlog 的原理限制，使用 TiDB Binlog 时，不会使用异步提交或一阶段提交特性。
+- 异步提交对写入量大的事务提升不明显。所以 TiDB 只在事务写入不超过 256 个键值对，所有键值对里键的总大小不超过 4 KB 时，才会使用异步提交或一阶段提交特性。
```
Comment (TomShawn) [inline faq/deploy-and-maintain-faq.md:237]:
```suggestion
- TiDB 只在事务写入不超过 256 个键值对，以及所有键值对里键的总大小不超过 4 KB 时，才会使用异步提交或一阶段提交特性。这是因为对于写入量大的事务，异步提交不能明显提升执行性能。
```

**Review by TomShawn — APPROVED** [review]
rest /lgtm

**Pull request merged**